### ISSUE TITLE
Remove API key on gem signout

### ISSUE THREAD

**Issue by jenshenny** [rubygems/rubygems.org#2844]
## Is your feature request related to a problem?

An API key is created on every `gem signin`. When `gem signout` is called, the API credentials are removed from the local machine, but are not removed from the rubygems.org host. This might be intended behaviour, but I would expect the API key to be removed on rubygems.org as it probably won't be used anymore.

## Describe the solution you'd like

On `gem signout`, delete the associated remote API key. This would be done in the rubygems repo.

**Comment by ecnelises** [comment]
This might be confusing because not every `gem signout` is paired with `gem signin`. Maybe we can set expiry to API keys so that a key unused for a long time will be automatically revoked.

**Comment by sonalkr132** [comment]
yeah, the user may have created the key from our web UI. The same API key may also be used in multiple environments.
What we can do is perhaps give a prompt or add a flag for deleting the key from the server-side as well.
When we use to have only one key per account, keeping it the same made more sense.

**Comment by jenshenny** [comment]
Closing as you can now add expiry to api keys https://github.com/rubygems/rubygems.org/pull/4681
